activetable.js

```diff
@@ -5,12 +5,35 @@ var ActiveTable = (function () {
     NUMERIC = 1;
     STRING = 2;
 
+    // Helper function to determine the number of significant digits in a
+    // floating point literal.  Significant digits are counted from the first
+    // non-zero digit specified, but include trailing zeros.  The string s is
+    // assumed to be a valid floating-point literal and is not validated.
+    function significant_digits(s) {
+        // Regular expression to capture the digits before and after the decimal
+        // point, but not the digits in the exponent.
+        var re = /([0-9]*)\.([0-9]+)|([0-9]+)/;
+        // All specified digits without the decimal point and the exponent.
+        var all_digits = re.exec(s).slice(1).join('');
+        // The number of digits with leading zero digits removed.
+        return all_digits.replace(/^0+/, '').length;
```

Comment:
This doesn't work either, since you also have to remove zeros after the decimal point if all digits before the decimal point are zero.  This could be collapsed into the regex in some way, but since this function is so incredibly prone to subtle bugs, I'd like to keep it as simple as possible.
(There are two additional problems with you regex: floating-point literals are allowed to have whitespace before and after the sign, and an explicit plus sign is also allowed).